### DISCUSSION TITLE
Python 3.11 is no longer available, can newer versions be supported?

### DISCUSSION THREAD

**Discussion by adubry75** [open-webui/open-webui#7650] [Ideas]
# Feature Request

**Is your feature request related to a problem? Please describe.**
Python 3.11 is no longer supported due to security vulnerabilities AND the python website no longer gives access to the binaries to download.

**Describe the solution you'd like**
Please make open-webui compatible with python 3.12 or higher.

**Describe alternatives you've considered**
Source code for python 3.11 is available but compiling it sounds like an extra step in pain.

**Additional context**
I will download the newest python and see if open-webui will install, even though the warnings when I tried to install it with 3.10 (my current version), it says:  Requires-Python <3.12.0a1,>=3.11

Hopefully that's just an old warning and maybe it will work until 3.12 or 3.13. I'll try that. If the documentation on the read.me or the warnings are old and outdated, please update :). Thanks for a great software!

**Comment by adubry75** [comment]
follow up: I subsequently tried to install it ignoring the version requirements:

pip install --ignore-requires-python open-webui

which worked ALMOST all the way but ran into several other dependency issues. Eventually I gave up trying to install it via python and just installed it with docker:

docker run -d --name open-webui -p 3000:8080 -v open-webui:/app/backend/data --restart=always ghcr.io/open-webui/open-webui:main

(after installing Docker for Windows).

I know python is terribly popular though, so others might run across this issue if they try the python install directions. feel free to close this discussion if it's not deemed worthy. Thanks! :)

**Comment by KertLynx** [comment]
> **Is your feature request related to a problem? Please describe.** Python 3.11 is no longer supported due to security vulnerabilities AND the python website no longer gives access to the binaries to download.

I don't think this is is true at all!  Python 3.11 is still supported and available for download from the official Python website. There are no indications that Python 3.11 has been discontinued due to security vulnerabilities, nor has access to its binaries been removed. There is no lack of official support and availability of Python 3.11 from the Python Software Foundation.

**Reply by wilson0x4d** [reply]
actually, 3.11 is only supported for "security updates" and is currently on its EOL track. this also means that keeping a secure deployment of 3.11 requires manually building from sources (updated installers are no longer made available for download.)

consequently, installing 3.11 anew requires installing from an older installer that does not have the latest security updates (or requires a developer to get involved.)

a move into 3.13 would be wise since it just released, and it will remain in active support for the next 2 years.

unless there is some version-specific dependency it would be nice to not be pinned to specific versions of Python.

if there are known issues preventing using to 3.13 or later it would be nice to have a link to a "tracking issue" that aggregates all of the Issue Numbers required to move beyond 3.11.

_EDIT: typos/clarity._

**Reply by nighthawk475** [reply]
Just wanted to come reiterate this.

To quote python.org's page about 3.11.15 (latest 3.11.x release)
It says specifically:
> No installers
> According to the release calendar specified in [PEP 664](https://peps.python.org/pep-0664/), Python 3.11 is now in the "security fixes only" stage of its life cycle: 3.11 branch only accepts security fixes and releases of those are made irregularly in source-only form until October 2027. Python 3.11 isn't receiving regular bug fixes anymore, and binary installers are no longer provided for it. Python 3.11.9 was the last full bugfix release of Python 3.11 with binary installers.

Having to choose between manually building your own python installation from the source code, or resigning yourself to an older insecure version, doesn't feel right for hosting a web server application that many users will open access to beyond just their local computer or lan. And regardless of security, it's just quite a bit of friction for newcomers who just want to experience OpenWebUI. The non-docker options no longer feel like a "It just works!" solution anymore, IMO.

Especially since none of this is mentioned anywhere in the OpenWebUI installation documentation. It doesn't even have a word about the python version requirements as far as I can see?

**Reply by Classic298** [reply]
Open webui docs DOES mention that 3.11-3.12 is supported only

We are still waiting on one last dependency i think langchain Community to introduce .13 support

**Reply by nighthawk475** [reply]
Ahh - perhaps in some places.

I just double checked now, and on the Python page of the Quick Start guide section of the docs it only mentions 3.11 as part of the example commands in the `uv` and `Conda` sub tabs, not on the default `pip` sub tab that the page loads with. (And even then it's just a part of the installation/execution command, but the fact it's actually a requirement with unusual installation steps required isn't explicitly stated)

If it's mentioned anywhere else in a more general way then that wasn't something I had run across on my way through the docs to set up my new installation recently.

Edit: also, 3.12.x  (up to 3.12.13)  *is* supported now? That was also something I hadn't seen in the installation section of the docs. Apologies for my being uninformed. (Though that doesn't particularly change the issue as 3.12 is also into security only releases that need a different flow for installation, like 3.11)

**Reply by Classic298** [reply]
good feedback ill make sure the version requirement is more prominent in the docs.

afaik 3.12.x is supported. but the most stable is definitely 3.12 - and 3.12.13 might work, SHOULD WORK, but best test it out a bit.

**Comment by rgaricano** [comment]
https://github.com/open-webui/open-webui/blob/b03fc97e287f31ad07bda896143959bc4413f7d2/pyproject.toml#L129-L137

**Comment by astr0gator** [comment]
Related: https://github.com/open-webui/open-webui/pull/7850#issuecomment-2750077165
